subxt/src/book/usage/events.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019-2022 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! # Events
+//!
+//! In the process of adding extrinsics to a block, they are executed. When extrinsics are executed,
+//! they may produce events describing what's happening, but additionally the node may add emit some
+//! events of its own as the block is processed. Events live in a single location in node storage
+//! which is overwritten at each block.
```

Comment:
If this storage is overwritten at each block, I would like to understand, why we are able to query events at a certain hash from the past then, and not only at the latest block. For example, `client.blocks().at(hash).await?.events().await?;` is possible, right? How does this work if the hash is from the past and the storage for the events is "overwritten"?

Reply:
Aaah okay, yes you are right we should not teach too much about the internals of substrate here. 
Maybe we can say:
"Extrinsics are executed in some pallet. In this process Events are emitted and stored in the node for some time, so it is possible to query for the events an extrinsic has caused. This is possible for a few blocks into the past, although in most cases you probably want to query for the events in the latest block with the function `at_latest()`"